dt-metrics/combined/date-range-activity.php

```diff
@@ -0,0 +1,370 @@
+        }
+        $params = $request->get_params();
+        if ( isset( $params['post_type'], $params['field_id'] ) ){
+
+            $post_type = $params['post_type'];
+            $field_id = $params['field_id'];
+
+            // Capture rendered field html.
+            ob_start();
+            $field_settings = DT_Posts::get_post_field_settings( $post_type );
+            $field_settings[$field_id]['custom_display'] = false;
+            render_field_for_display( $field_id, $field_settings, null );
+            $rendered_field_html = ob_get_contents();
+            ob_end_clean();
+
+            return [
+                'html' => $rendered_field_html
+            ];
+        }
+
+        return [];
+    }
+
+    public function dummy_endpoint( WP_REST_Request $request ){
+        return new WP_REST_Response( [] );
+    }
+
+    public function date_range_activity( WP_REST_Request $request ){
+        if ( !$this->has_permission() ){
+            wp_send_json_error( new WP_Error( 'date_range_activity', 'Missing Permissions', [ 'status' => 400 ] ) );
+        }
+
+        $params = $request->get_params();
+        if ( isset( $params['post_type'], $params['field'], $params['ts_start'], $params['ts_end'] ) ){
+
+            // Fetch associated field settings.
+            $settings = $this->get_field_settings( $params['post_type'] )[$params['field']];
+            $field_type = $settings['type'];
+
+            // Proceed with generating corresponding select SQL parts.
+            $field_type_sql = "AND field_type = '" . esc_sql( $field_type ) . "'";
+            $meta_key_sql = "AND meta_key LIKE '" . esc_sql( $params['field'] ) . "'";
+            $obj_subtype_sql = '';
+
+            // Accommodate special cases.
+            if ( $field_type == 'communication_channel' ){
+                $field_type_sql = "AND (field_type = '' OR field_type = '" . esc_sql( $field_type ) . "')";
+                $meta_key_sql = "AND meta_key LIKE '" . esc_sql( $params['field'] ) . "%'";
+                $meta_value_sql = "AND meta_value LIKE '" . ( empty( $params['value'] ) ? '%' : esc_sql( $params['value'] ) ) . "'";
+
+            } elseif ( ( $field_type == 'connection' ) || ( $field_type == 'location' ) ){
+                $meta_key_sql = ( $field_type == 'connection' ) ? "AND meta_key LIKE '%'" : "AND meta_key LIKE '" . esc_sql( $params['field'] ) . "'";
+
+                $values = [];
+                foreach ( $params['value'] ?? [] as $value ){
+                    if ( isset( $value['ID'] ) ){
+                        $values[] = $value['ID'];
+                    }
+                }
+                $meta_value_sql = ( !empty( $values ) ? 'AND meta_value IN (' . dt_array_to_sql( $values ) . ')' : "AND meta_value LIKE '%'" );
+                $obj_subtype_sql = ( $field_type == 'connection' ) ? "AND object_subtype = '" . esc_sql( $params['field'] ) . "'" : '';
```

Comment:
@kodinkat object subtype used to be 'p2p', so this does not pick up on historical data.
<img width="752" alt="image" src="https://github.com/DiscipleTools/disciple-tools-theme/assets/24901539/0f6d235c-88ec-430c-ac33-93b5d7996f57">

Reply:
Historical `p2p` activity entries now captured; however, please note that any `p2p` entries detected within given date range, will be included; as there is no apparent way to filter by field id.

Reply:
@kodinkat I think there are 2options here
We use the meta_key and the field type:
<img width="600" alt="image" src="https://github.com/DiscipleTools/disciple-tools-theme/assets/24901539/4ea4c26c-02b9-48d5-befe-62bcdcb2bc28">


Or we write a migration to upgrade the old activity to match the new shape.

Reply:
@corsacca yep, maybe the first option, re exploring the use of both the meta_key and field_type, to determine actual field_id.... As the 2nd option sounds like a new ticket? ;)

Reply:
Yeah. Let's go for #1 and see where we get.

Reply:
@corsacca does the latest sql tweak hold true for you?

Reply:
@kodinkat almost. 
We do need to add the field type. Other wise it gets both the  `Baptized` and the `baptized by` field.
<img width="1283" alt="image" src="https://github.com/DiscipleTools/disciple-tools-theme/assets/24901539/b76aaa59-9c6d-42b4-ae2e-c956f9603847">

Safir:
![Uploading image.png…]()